src/sagas.js

```diff
@@ -646,6 +656,23 @@ function* fetchVmNics ({ vmId }) {
   return []
 }
 
+function* fetchVmsSnapshots ({ vms }) {
+  yield all(vms.map((vm) => call(function *() {
+    const snapshotsInternal = yield fetchVmSnapshots({ vmId: vm.id })
+    yield put(setVmSnapshots({ vmId: vm.id, snapshots: snapshotsInternal }))
+  })))
+}
+
+export function* fetchVmSnapshots ({ vmId }) {
```

Comment:
This saga is called from 3 places and each call is immediately followed by storing fetched snapshot to redux. Can we move invocation of `setVmSnapshots` into this saga?